text/0057-single-field-shortcut.md

```diff
@@ -0,0 +1,167 @@
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+* A novice might not understand that it's possible to create a `Register` with multiple `Fields` if exposed only to code which uses single-`Field` `Register`s.
+* There is a difference between a `Register` with one named/indexed `Field` and with one unnamed `Field`.
+* Library code will be made more confusing by having `fields` refer to possibly only one field.
+* Downstream code will have to deal with a third type of object returned by the
+  `.f` property which behaves unlike the current two.
+* BSP/doc generators will have to intelligently render a `Field` with an empty name and path (likely by omitting them, possibly by reusing the register's name).
+    * An empty name is prohibited by `FieldActionMap`, so empty names are a novel
+      problem.
+    * The register does not know its own name independently of its container, so this may be difficult in practice.
+* Adding a second `Field` to a register created this way will break all accesses to the `.f` property and likely all BSP users (though the actual bus transactions to the first `Field` will not change).
+    * This could be especially entertaining if the new `Field` names happen to
+      be the same as properties of the first `Field`'s `FieldAction`.
+    * BSP users will have to change all constants/functions named after just the
+      register to instead use the register + the field name.
+
+## Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+* Simple and easily explainable change, easy to migrate code to, though no
+  migration is even necessary
+* Minimal change to the library's existing code
+* Nicely reduces redundant names, types, and properties
+* Not strictly necessary, not having it will just make peripheral code a little
+  more cluttered
+* Users could instead be taught to eliminate the subclass by passing a dict to
+  the `Register` constructor and assign the `.f` property of interest to a
+  local variable to de-clutter code
+
+## Prior art
+[prior-art]: #prior-art
+
+AVR microcontrollers have many registers where the field name is the same as the register name, though this is not always the case for single-field registers. Application code usually uses the register name in this case and ignores the field name.
+
+STM32 microcontrollers are similar, but the field name is usually a suffix of the register name, as the field name is not expected to be globally unique. However, the generated BSP files do appear to always contain the field name at least somewhere.
+
+More experienced input is welcomed here.
+
+## Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+* Should we change `fields` to `field` in the constructor and properties? This could break existing code.
```

Comment:
Basically everyone uses `.f`, but yeah.

Reply:
I think it's more likely to be a problem with the constructor argument.

Reply:
Breakage isn't an issue yet (until 0.1 is released).

I see three possibilities:
a) do nothing;
b) add a `.field` property, aliased to `.fields`;
c) rename `.fields` to `.field`, but keep the `fields=` parameter;
d) do c) and rename `fields=` to `field=`;

Option c) has my preference.

Considering the example given in this RFC, a single-field register won't pass `fields=` as a keyword argument (`csr.Register(csr.Field(...), access="rw")`), so having it plural is mostly invisible.

OTOH the plural `.fields` accessor is awkward for a single-field register.

Reply:
We settled that the property will be `.field` and the constructor will be `fields=`.